flutter/lib/src/sentry_flutter.dart

```diff
@@ -62,8 +55,8 @@ mixin SentryFlutter {
   ) async {
     options.debug = kDebugMode;
 
-    // web still uses a http transport for Web which is set by default
-    if (!kIsWeb) {
+    // Not all platforms have a native integration.
+    if (options.platformChecker.hasNativeIntegration) {
```

Comment:
`hasNativeIntegration` will need to be changed pretty soon, maybe we do `isAndroid or `isIOS`?

Reply:
Even though we'll need to change it, If we're checking for "having a native SDK" across this code base in order to verify things like transport etc, it might be a good idea to have this single place. It'll make it easier to to change only its implementation later, when we add a native layer for example to macOS or Windows.

Also it communicating intent a bit better than if (ios or android)